tests/autopilot-capacity/aut_test.go

```diff
@@ -33,71 +95,128 @@ var _ = BeforeSuite(func() {
+// PVC sizes of the volumes
+var _ = Describe(fmt.Sprintf("{%sWaitForWorkload}", testNameSuite), func() {
 	It("has to fill up the volume completely, resize the volume, validate and teardown apps", func() {
 		var contexts []*scheduler.Context
 		var err error
 		testName := strings.ToLower(fmt.Sprintf("%sBasic", testNameSuite))
 
-		apParameters := &scheduler.AutopilotParameters{
-			Enabled: true,
-			Name:    testName,
-			AutopilotRuleParameters: scheduler.AutopilotRuleParameters{
-				ActionsCoolDownPeriod: 60,
-				PVCWorkloadSize:       10737418240, //10Gb
-				PVCPercentageUsage:    50,
-				PVCPercentageScale:    50,
-			},
-		}
-
-		Step("schedule applications", func() {
-			for i := 0; i < Inst().ScaleFactor; i++ {
-				taskName := fmt.Sprintf("%s-%v", fmt.Sprintf("%s-%d", testName, i), Inst().InstanceID)
-				context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
-					AppKeys:             Inst().AppList,
-					StorageProvisioner:  Inst().Provisioner,
-					AutopilotParameters: apParameters,
-				})
-				Expect(err).NotTo(HaveOccurred())
-				Expect(context).NotTo(BeEmpty())
-				contexts = append(contexts, context...)
+		for _, apRule := range autopilotruleBasicTestCases {
+			apParameters := &scheduler.AutopilotParameters{
+				Enabled:                 true,
+				Name:                    testName,
+				AutopilotRuleParameters: apRule,
 			}
-		})
 
-		Step("wait until workload completes on volume", func() {
-			for _, ctx := range contexts {
-				err = Inst().S.WaitForRunning(ctx, workloadTimeout, retryInterval)
-				Expect(err).NotTo(HaveOccurred())
-			}
-		})
-
-		Step("validating volumes and verifying size of volumes", func() {
-			for _, ctx := range contexts {
-				err = Inst().S.InspectVolumes(ctx, timeout, retryInterval)
-				Expect(err).NotTo(HaveOccurred())
-			}
-		})
+			Step("schedule applications", func() {
+				for i := 0; i < Inst().ScaleFactor; i++ {
+					taskName := fmt.Sprintf("%s-%v", fmt.Sprintf("%s-%d", testName, i), Inst().InstanceID)
+					context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+						AppKeys:             Inst().AppList,
+						StorageProvisioner:  Inst().Provisioner,
+						AutopilotParameters: apParameters,
+					})
+					Expect(err).NotTo(HaveOccurred())
+					Expect(context).NotTo(BeEmpty())
+					contexts = append(contexts, context...)
+				}
+			})
+
+			Step("wait until workload completes on volume", func() {
+				for _, ctx := range contexts {
+					err = Inst().S.WaitForRunning(ctx, workloadTimeout, retryInterval)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+
+			Step("validating volumes and verifying size of volumes", func() {
+				for _, ctx := range contexts {
+					err = Inst().S.InspectVolumes(ctx, timeout, retryInterval)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+
+			Step(fmt.Sprintf("wait for unscheduled resize of volume (%s)", unscheduledResizeTimeout), func() {
+				time.Sleep(unscheduledResizeTimeout)
```

Comment:
we can do this just once after finishing all `autopilotruleBasicTestCases`. This will save a bunch of time instead of sleeping here for each test case.

Reply:
No we can't do this once.if we create multiple autopilot rules and schedule the sane apps, they will apply for all PVC at the same time.

Reply:
Synced up offline on this. To so this, we will need to change test to schedule each test case in a different namespace